src/types/Cache.ts

```diff
@@ -0,0 +1,15 @@
+import { SecondLevelDomain } from "./SecondLevelDomain";
+import { TopLevelDomain } from "./TopLevelDomain";
+
+export interface ICache {
```

Comment:
All of these methods should return promises to support network (database, redis, etc..) caches.